lib/internal/modules/cjs/loader.js

```diff
@@ -1191,33 +1191,16 @@ Module.prototype._compile = function(content, filename) {
 };
 
 // Native extension for .js
-let warnRequireESM = true;
 Module._extensions['.js'] = function(module, filename) {
   if (filename.endsWith('.js')) {
     const pkg = readPackageScope(filename);
+    // Function require shouldn't be used in ES modules.
     if (pkg && pkg.data && pkg.data.type === 'module') {
-      if (warnRequireESM) {
-        const parentPath = module.parent && module.parent.filename;
-        const basename = parentPath &&
-            path.basename(filename) === path.basename(parentPath) ?
-          filename : path.basename(filename);
-        process.emitWarning(
-          'require() of ES modules is not supported.\nrequire() of ' +
-          `${filename} ${parentPath ? `from ${module.parent.filename} ` : ''}` +
-          'is an ES module file as it is a .js file whose nearest parent ' +
-          'package.json contains "type": "module" which defines all .js ' +
-          'files in that package scope as ES modules.\nInstead rename ' +
-          `${basename} to end in .cjs, change the requiring code to use ` +
-          'import(), or remove "type": "module" from ' +
-          `${path.resolve(pkg.path, 'package.json')}.`,
-          undefined,
-          undefined,
-          undefined,
-          true
-        );
-        warnRequireESM = false;
-      }
-      throw new ERR_REQUIRE_ESM(filename);
+      const parentPath = module.parent && module.parent.filename;
+      const packageJsonPath = path.resolve(pkg.path, 'package.json');
+      const basename = path.basename(filename) === path.basename(parentPath) ?
```

Comment:
Since `basename` is derivable from `filename` and `parentPath`, it may make sense to have this handled within the error code...

Reply:
How do i get basename without using path module?

Reply:
No problem at all, thanks for working on this.

So there's nothing wrong with having `path` in use in the errors code. Either statically or for the lazy option just having `const path = require('path')` within the error function itself is also fine.

Reply:
Just a annotation: for any reason I don't know, errors code file throws just by using `require('path')`, using that lazy method fixed that issue.